### PR TITLE
Deepbgc

### PR DESCRIPTION
**Pull request by louperelo**
deepBGC is run with .fasta input. It then runs it's own Prodigal version. But funcscan's  Prodigal-process is called anyway ... I did not figure out how to modify the huge "if" statement in funcscan.nf in order to avoid this. 
To me it looked like it should not be run, if the command is:
`nextflow run . -profile docker --run_bgc_screening true --bgc_skip_antismash true --bgc_skip_gecco true --bgc_skip_hmmsearch true --outdir results/deepbgctest --input assets/samplesheet.csv`
and either prokka or prodigal is run if:
`if ( ( params.run_arg_screening && !params.arg_skip_deeparg ) || ( params.run_amp_screening && ( !params.amp_skip_hmmsearch || !params.amp_skip_amplify || !params.amp_skip_ampir ) ) || ( params.run_bgc_screening && ( !params.amp_skip_hmmsearch || !params.bgc_skip_antismash ) ) )`

closes issue #116 
## PR checklist

- [ x] This comment contains a description of changes (with reason).
- [x ] If you've fixed a bug or added code that should be tested, add tests!
  - [ x] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/funcscan/tree/master/.github/CONTRIBUTING.md)
  - [ ] If necessary, also make a PR on the nf-core/funcscan _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ x] Make sure your code lints (`nf-core lint`).
- [ x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [x ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by jfy133** [comment]
> But funcscan's Prodigal-process is called anyway ... I did not figure out how to modify the huge "if" statement in funcscan.nf in order to avoid this.

To avoid what exactly?

**Comment by louperelo** [comment]
If I only run the bcc workflow and only use deepbgc, I don't need prodigal to run

**Comment by jasmezz** [comment]
Note: If the input is not `.fasta` but `.gbk`, DeepBGC recognizes that as annotated sequences and skips its internal prodigal step. Maybe that is what we should go for. I remember the raw prodigal output not being recognized by DeepBGC, so we would need to enforce prokka if `--bgc_skip_deepbgc` is false. I can have a look later today!

**Comment by louperelo** [comment]
I decided going for the `.fasta` input, because I could not really see an advantage in using the `.gbk`. 
If the default setting `--run_annotation_tool prodigal` is used, we would have to run it twice anyway to once generate the `.faa` file and in a second run the `.gbk` file (Prodigal does not have the option to emit both in one run).  Also, I tried to use the `prokka.out.gbk` as `deepARG` input but without success. Would have to still find out why that wasn't accepted by `deepBGC`. 
Therefore, just using the `.fasta` input and running the internal `prodigal` in `deepBGC` seemed to be the easiest way to get a working version.

File: modules.json [hunk 1]
```diff
@@ -45,6 +45,14 @@
                     "git_sha": "15c7190e2271de7ff347940460a484f69e27a106",
                     "branch": "master"
                 },
+                "deepbgc/download": {
+                    "git_sha": "eae945721d9ac32227b95350b0c6d39a00d8a14f"
```
Comment (louperelo) [inline modules.json:49]:
```suggestion
                    "git_sha": "eae945721d9ac32227b95350b0c6d39a00d8a14f",
```

File: modules.json [hunk 2]
```diff
@@ -45,6 +45,14 @@
...
+                    "branch": "master"
+                },
+                "deepbgc/pipeline": {
+                    "git_sha": "eae945721d9ac32227b95350b0c6d39a00d8a14f"
```
Comment (louperelo) [inline modules.json:53]:
```suggestion
                    "git_sha": "eae945721d9ac32227b95350b0c6d39a00d8a14f",
```

File: modules.json (hunk 1 above)
Comment (louperelo) [inline modules.json:49]:
```suggestion
                    "git_sha": "eae945721d9ac32227b95350b0c6d39a00d8a14f",
```

File: workflows/funcscan.nf [hunk 3]
```diff
@@ -133,19 +133,22 @@ workflow FUNCSCAN {
...
             ch_annotation_gff        = PROKKA.out.gff
+            ch_annotation_gbk        = PROKKA.out.gbk
         }
 
     } else {
 
         ch_annotation_faa        = Channel.empty()
         ch_annotation_fna        = Channel.empty()
         ch_annotation_gff        = Channel.empty()
+        ch_annotation_gbk        = Channel.empty()
```
Comment (jasmezz) [inline workflows/funcscan.nf:151]:
```suggestion
        }   else if ( params.run_annotation_tool == "prokka") {
            PROKKA ( ch_prepped_input, [], [] )
            ch_versions              = ch_versions.mix(PROKKA.out.versions)
            ch_annotation_faa        = PROKKA.out.faa
            ch_annotation_fna        = PROKKA.out.fna
            ch_annotation_gff        = PROKKA.out.gff
        }

    } else {

        ch_annotation_faa        = Channel.empty()
        ch_annotation_fna        = Channel.empty()
        ch_annotation_gff        = Channel.empty()
```
We don't use GBK right now, so I removed those 3 lines.

**Review by jasmezz — CHANGES_REQUESTED** [review]
Looks almost good to me. Database output should go into `databases/` and not `bgc/deepbgc/db/`.

File: CITATIONS.md [hunk 4]
```diff
@@ -34,6 +34,10 @@
 
   > Arango-Argoty, Gustavo, Emily Garner, Amy Pruden, Lenwood S. Heath, Peter Vikesland, and Liqing Zhang. 2018. "DeepARG: A Deep Learning Approach for Predicting Antibiotic Resistance Genes from Metagenomic Data." Microbiome 6 (1): 23.
 
+- [DeepBGC](https://doi.org/10.1093/nar/gkz654)
+
+  > Hannigan, Geoffrey D, David Prihoda, Andrej Palicka, Jindrich Soukup, Ondrej Klempir, Lena Rampula, Jindrich Durcak, Michael Wurst, Jakub Kotowski, Dan Chang, Rurun Wang, Grazia Piizzi, Gergely Temesi, Daria J Hazuda, Christopher H Woelk, Danny A Bitton, A deep learning genome-mining strategy for biosynthetic gene cluster prediction, Nucleic Acids Research, Volume 47, Issue 18, 10 October 2019, Page e110
```
[the remark this answers is not in the document]
Comment (louperelo) [reply CITATIONS.md:39]: I tried to stick to the format of the previous citations: first author many times is <surname, first name> and all co-authors the other way round. 
But, yes, definitely, we should review this and give it all the same format!

**Review by jfy133 — CHANGES_REQUESTED** [review]
Just missing the prodgial version module update I think?

**Comment by jfy133** [comment]
@nf-core-bot fix linting

**Pull request merged**